site/en/blog/inside-the-container-query-polyfill/index.md

```diff
@@ -0,0 +1,301 @@
+---
+title: "Inside the container query polyfill"
+description: "Learn how the container query polyfill brings container query support to older browsers."
+layout: 'layouts/blog-post.njk'
+date: 2022-10-01
+hero: 'image/0Nc53fqpS4cgxOG9PCAyLwS0QwB3/75EKsp0A2SyU8bOWttSd.jpg'
+alt: "An assortment of neatly arranged drafting tools. In the center is a blank sheet of graphing paper."
+tags:
+  - css
+authors:
+  - gbmonaco
+---
+
+[Container queries](https://www.youtube.com/watch?v=gCNMyYr7F6w) are a new CSS feature that enables you to write styling logic that targets features of a parent element (for example, its width or height) to style its children. Recently, a [big update](https://developer.chrome.com/blog/cq-polyfill/) to the [polyfill](https://github.com/GoogleChromeLabs/container-query-polyfill) was released, coinciding with support landing in browsers.
+
+In this post, you will be able to take a peek inside how the polyfill works, the challenges it overcomes, and the best practices when using it to provide a great user experience for your visitors.
+
+## Under the hood
+
+### Transpilation
+
+When the CSS parser inside a browser encounters an unknown at-rule, like the brand new `@container` rule, it will simply discard it as if it never existed. Therefore, the first and most important thing the polyfill must do is transpile an `@container` query into something that won’t be discarded.
+
+{% Aside %}
+This sort of transpilation often occurs at build-time, as part of a post-processing step. However, the polyfill does this at runtime instead, for two reasons: you and your visitors. With support for container queries already in [Chromium 105](https://developer.chrome.com/blog/new-in-chrome-105/) and [Safari 16](https://webkit.org/blog/13152/webkit-features-in-safari-16-0/), it’s expected that most of your visitors will likely be running browsers with built-in support sooner rather than later. Offline transpilation would effectively tax you (for having to maintain this offline post-processing step) _and_ your visitors (for having to download transpiled code that may never be used).
```

Comment:
```suggestion
This sort of transpilation often occurs at build-time, as part of a post-processing step. However, the polyfill does this at runtime instead, for two reasons: you and your visitors. With support for container queries already in [Chromium 105](/blog/new-in-chrome-105/) and [Safari 16](https://webkit.org/blog/13152/webkit-features-in-safari-16-0/), it’s expected that most of your visitors will likely be running browsers with built-in support sooner rather than later. Offline transpilation would effectively tax you (for having to maintain this offline post-processing step) _and_ your visitors (for having to download transpiled code that may never be used).
```